test/common/quic/envoy_quic_client_session_test.cc

```diff
@@ -73,45 +77,50 @@ class EnvoyQuicClientSessionTest : public testing::TestWithParam<quic::ParsedQui
             *Network::Test::getCanonicalLoopbackAddress(TestEnvironment::getIpVersionsForTest()[0]),
             54321)),
         peer_socket_(createConnectionSocket(self_addr_, peer_addr_, nullptr)),
-        quic_connection_(new TestEnvoyQuicClientConnection(
-            quic::test::TestConnectionId(), connection_helper_, alarm_factory_, writer_,
-            quic_version_, *dispatcher_, createConnectionSocket(peer_addr_, self_addr_, nullptr),
-            connection_id_generator_)),
         crypto_config_(std::make_shared<quic::QuicCryptoClientConfig>(
             quic::test::crypto_test_utils::ProofVerifierForTesting())),
         quic_stat_names_(store_.symbolTable()),
         transport_socket_options_(std::make_shared<Network::TransportSocketOptionsImpl>()),
-        envoy_quic_session_(
-            quic_config_, quic_version_,
-            std::unique_ptr<TestEnvoyQuicClientConnection>(quic_connection_),
-            quic::QuicServerId("example.com", 443, false), crypto_config_, *dispatcher_,
-            /*send_buffer_limit*/ 1024 * 1024, crypto_stream_factory_, quic_stat_names_, {},
-            *store_.rootScope(), transport_socket_options_, {}),
         stats_({ALL_HTTP3_CODEC_STATS(POOL_COUNTER_PREFIX(store_, "http3."),
                                       POOL_GAUGE_PREFIX(store_, "http3."))}) {
     http3_options_.mutable_quic_protocol_options()
         ->mutable_num_timeouts_to_trigger_port_migration()
         ->set_value(1);
+  }
+
+  void SetUp() override {
+    quic_connection_ = new TestEnvoyQuicClientConnection(
+        quic::test::TestConnectionId(), connection_helper_, alarm_factory_, writer_, quic_version_,
+        *dispatcher_, createConnectionSocket(peer_addr_, self_addr_, nullptr, /*prefer_gro=*/true),
+        connection_id_generator_);
+
+    OptRef<Http::HttpServerPropertiesCache> cache;
+    OptRef<Network::UpstreamTransportSocketFactory> uts_factory;
+    envoy_quic_session_ = std::make_unique<EnvoyQuicClientSession>(
+        quic_config_, quic_version_,
+        std::unique_ptr<TestEnvoyQuicClientConnection>(quic_connection_),
+        quic::QuicServerId("example.com", 443, false), crypto_config_, *dispatcher_,
+        /*send_buffer_limit*/ 1024 * 1024, crypto_stream_factory_, quic_stat_names_, cache,
+        *store_.rootScope(), transport_socket_options_, uts_factory);
```

Comment:
NOTE: I had to move this initialization to `SetUp()` so I could change the allow_mmsg value and have it picked up in the construction of these objects.  That also required making EnvoyQuicClientSession a unique_ptr, since it doesn't have a copy constructor.